api/src/main/java/org/pmiops/workbench/db/model/Workspace.java

```diff
@@ -300,12 +307,32 @@ public void setContainsUnderservedPopulation(Boolean containsUnderservedPopulati
   @ElementCollection(fetch = FetchType.LAZY)
   @CollectionTable(name = "underserved_populations", joinColumns = @JoinColumn(name = "workspace_id"))
   @Column(name = "underserved_population")
-  public Set<UnderservedPopulationEnum> getUnderservedPopulationSet() {
+  public Set<Short> getUnderservedPopulationsShort() {
     return underservedPopulationSet;
   }
 
-  public void setUnderservedPopulationSet(Set<UnderservedPopulationEnum> newUnderservedPopulationSet) {
-    this.underservedPopulationSet = newUnderservedPopulationSet;
+  public void setUnderservedPopulationsShort(Set<Short> newUnderservedPopulations) {
+    this.underservedPopulationSet = newUnderservedPopulations;
+  }
+
+  @Transient
+  public Set<UnderservedPopulationEnum> getUnderservedPopulationsEnum() {
+    Set<Short> from = getUnderservedPopulationsShort();
```

Comment:
If this were an `Optional`, the if condition could probably be simplified.

Reply:
Tried this but unfortunately I need the annotated getter to have a type that can be mapped to the stored type, else Spring Data JPA cannot map it to the SQL column. If we wanted an `Optional` version of this method, I'd need to add another method and make it `@Transient` as well which would probably not be worth it.

https://vladmihalcea.com/the-best-way-to-map-a-java-1-8-optional-entity-attribute-with-jpa-and-hibernate/

Reply:
Great article, thanks, and yeah, more effort than it saves.